src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx

```diff
@@ -493,6 +504,18 @@ const CheckoutViewWrapper = () => {
     setAuthMethod('signUp');
   };
 
+  const recalculatePrice = async (
+    priceId: string,
+    currency: string,
+    promoCodeName?: string,
+    postalCode?: string,
+    country?: string,
+  ) => {
+    const price = await checkoutService.getPriceById({ priceId, currency, promoCodeName, postalCode, country });
+    setSelectedPlan(price);
+    setPlan(price);
```

Comment:
It is a bit confusing see `price` named variable adding it to a `plan` ones 😵‍💫

Reply:
Yes, I know. We have to remove useless code in general... It was used when we had upsell plans, but since we only have yearly plans, we don't need them anymore... anyways, I will try to remove this state.